lib/matplotlib/tests/test_backend_pdf.py

```diff
@@ -71,6 +71,17 @@ def test_multipage_pagecount():
         assert pdf.get_pagecount() == 2
 
 
+def test_multipage_properfinalize():
+    pdfio = io.BytesIO()
+    with PdfPages(pdfio) as pdf:
+        for i in range(10):
+            fig = plt.figure()
+            ax = fig.add_subplot(111)
+            fig.savefig(pdf, format="pdf")
+    pdfio.seek(0)
+    assert sum(b'Kids' in line for line in pdfio) == 1
```

Comment:
The PDF specification recommends arranging the page objects in a tree with `Kids` and `Parent` pointers, for fast retrieval. Currently we just make a one-level tree, but if this is ever fixed, it would break this test. How about checking for `startxref` instead?

Reply:
I'm happy to punt improving this test to be part of that work.

Reply:
I'm fine with merging this as is, but I think changing `'Kids'` to `'startxref'` here would still be correct and would be more robust against future changes.

Reply:
I'm fine w that, but as below, what about just checking the size is small?

Reply:
Oh, I miss-understood the suggestion, thought `'startxref'` was some sort of xml parsing or something (sorry not enough coffee yet).

:+1: to changing the search term to `'startxref'`